### PR TITLE
Bump knative to enable new features

### PR DESCRIPTION
**Pull request by pierretasci**
# Changes

This PR is related to https://github.com/tektoncd/pipeline/pull/3156 and splits out the knative version bump from that PR. Knative has some internal changes that we would like to take advantage of like the ability to configure the QPS it accepts which necessitates this version bump.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
knative is upgraded to a more recent version
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,35 +3,28 @@ module github.com/tektoncd/pipeline
...
 	github.com/tektoncd/plumbing v0.0.0-20200430135134-e53521e1d887
 	go.opencensus.io v0.22.4
 	go.uber.org/zap v1.15.0
 	golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
-	golang.org/x/text v0.3.3 // indirect
 	gomodules.xyz/jsonpatch/v2 v2.1.0
-	google.golang.org/api v0.25.0
-	k8s.io/api v0.17.6
-	k8s.io/apimachinery v0.17.6
+	k8s.io/api v0.18.7-rc.0
```
Comment (vdemeester) [inline go.mod:22]: This one is weird, it should be `v0.18.6`

File: go.mod [hunk 2]
```diff
@@ -3,35 +3,28 @@ module github.com/tektoncd/pipeline
...
+	k8s.io/apimachinery v0.19.0
```
Comment (vdemeester) [inline go.mod:23]: For consistency, this should be at the same level of the others, so `v0.18.6`

File: go.mod [hunk 3]
```diff
@@ -3,35 +3,28 @@ module github.com/tektoncd/pipeline
...
 	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
-	k8s.io/code-generator v0.18.0
+	k8s.io/code-generator v0.18.6
 	k8s.io/kube-openapi v0.0.0-20200410145947-bcb3869e6f29
-	knative.dev/pkg v0.0.0-20200702222342-ea4d6e985ba0
+	knative.dev/pkg v0.0.0-20200831162708-14fb2347fb77
```
Comment (vdemeester) [inline go.mod:27]: fwiw, this is `release-0.17`, so we could update `./hack/update-deps.sh`

```diff
diff --git a/hack/update-deps.sh b/hack/update-deps.sh
index 2d6d9cf5..1c6678dc 100755
--- a/hack/update-deps.sh
+++ b/hack/update-deps.sh
@@ -22,7 +22,7 @@ source $(git rev-parse --show-toplevel)/vendor/github.com/tektoncd/plumbing/scri
 
 cd ${REPO_ROOT_DIR}
 
-VERSION="release-0.16"
+VERSION="release-0.17"
 
 # The list of dependencies that we track at HEAD and periodically
 # float forward in this repository.
```

Comment (vdemeester) [reply go.mod:27]: `./hack/update-deps --upgrade` does a little bit more than that diff but that's alright.

**Review by vdemeester — COMMENTED** [review]
So one thing to note is that, the `k8s.io/*` in the diff do not really affect the k8s as it is override in the `replace` below. That's not a big deal as it does compile. We may want to bump k8s in another PR.

Otherwise, looking good :+1:

File: go.mod (hunk 1 above)
Comment (nishpa214) [reply go.mod:22]: Yeah, `k8s.io/*` in go.mod updated as part of updating the `knative/pkg` which also has same `rc.0` version.  See: https://github.com/knative/pkg/blob/master/go.mod#L48.

File: go.mod (hunk 3 above)
Comment (nishpa214) [reply go.mod:27]: Done!

**Review by vdemeester — COMMENTED** [review]
/lgtm

**Comment by bobcatfish** [comment]
thanks for taking this on @pierretasci !!

**Comment by ImJasonH** [comment]
/lgtm

Seems fairly straightforward.

cc @mattmoor

**Comment by mattmoor** [comment]
As a heads up, we switched to K8s 1.18 libs last week for our upcoming `0.18` release.  This is notable because K8s made breaking changes to the client-go APIs.  When y'all are ready to pick that up, feel free to ping me and I can help support, but that's largely orthogonal to this.

Glad this seemed so easy 🤩  😅

**Review by vdemeester — APPROVED** [review]
/meow

**Pull request merged**